src/Integration.Vsix/InfoBar/InfoBarManager.cs

```diff
@@ -70,26 +70,21 @@ public IInfoBar AttachInfoBar(Guid toolWindowGuid, string message, SonarLintImag
             return AttachInfoBarImpl(toolWindowGuid, message, null, imageMoniker);
         }
 
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("Reliability",
-            "S3215:\"interface\" instances should not be cast to concrete types",
-            Justification = "We want to hide the concrete implementation but also handle instance that were created by this class in the first place",
-            Scope = "member",
-            Target = "~M:SonarLint.VisualStudio.Integration.Vsix.InfoBar.InfoBarManager.DetachInfoBar(SonarLint.VisualStudio.Integration.InfoBar.IInfoBar)")]
         public void DetachInfoBar(IInfoBar currentInfoBar)
         {
             if (currentInfoBar == null)
             {
                 throw new ArgumentNullException(nameof(currentInfoBar));
             }
 
-            PrivateInfoBarWrapper wrapper = currentInfoBar as PrivateInfoBarWrapper;
-            if (wrapper == null)
+            if (!(currentInfoBar is PrivateInfoBarWrapper wrapper))
             {
                 throw new ArgumentException(Strings.InvalidInfoBarInstance, nameof(currentInfoBar));
             }
 
             currentInfoBar.Close();
 
+            ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
There are two possible paths through the code and I've added the thread helper checker in both places.
However, it hasn't stopped the VSX analyzer complaining, though - there are still half-a-dozen other analysis issues in this file, even though they are in lower-level methods that are guarded by a check in an upstream method.

Reply:
Is the `currentInfoBar.Close();` safe to run on non-UI thread?